lib/internal/url.js

```diff
@@ -817,6 +817,9 @@ class URLSearchParams {
     if (arguments.length < 1) {
       throw new TypeError('The `callback` argument needs to be specified');
     }
+    if (typeof callback !== 'function') {
```

Comment:
Our error messages don't even seem to match up though. I'd still change it, but I won't fight it too much.